### PR TITLE
Feature/cdf/ellipse

### PR DESCRIPTION
**Pull request by schneiderjan**
Thank you for your contribution to the **Tomni** package.
Before submitting/accepting this Pull Request, please make sure:

## Contributor of code

- [ ] Version number is updated
- [x] `HISTORY.rst` is updated
- [x] Function is tested
- [x] `(docstring)` is added or updated
- [ ] All needed package are in [requirements.txt](requirements.txt)

**For new function**

- [x] Functions are named based on [Python naming conventions](https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html)
- [ ] Function is added to apidocx.rst
- [ ] New feature is added to `README.md` subsection `Features`
- [x] New function is imported in all `__init__.py` that are needed

**For bugfix**
- [ ] Unit test is added that covers the bug (if possible)
- [ ] You cursed at the person that introduced the bug

## Reviewer

- [ ] Version number is updated
- [ ] `HISTORY.rst` is updated
- [ ] Function is tested
- [ ] `(docstring)` is added or updated
- [ ] All needed package are in [requirements.txt](requirements.txt)

**For new function**

- [ ] Functions are named based on [Python naming conventions](https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html)
- [ ] Function is added to apidocx.rst
- [ ] New feature is added to `README.md` subsection `Features`
- [ ] New function is imported in all `__init__.py` that are needed

**For bugfix**
- [ ] Unit test is added that covers the bug (if possible)

### CODE REVIEW

File: tomni/cytosmart_data_format/annotations/ellipse/main.py [hunk 1]
```diff
@@ -11,32 +11,89 @@ def __init__(self, radius: Point, center: Point, rotation: float):
...
+
+        # self._roundness = None # Needed?
+        # or this is also eccentricity?
 
     @property.getter
     def radius(self) -> Point:
-        """x and y
-        """
-        pass
+        return self._radius
```
[the remark this answers is not in the document]
Comment (schneiderjan) [reply tomni/cytosmart_data_format/annotations/ellipse/main.py:23]: It does have radiuses x and y.

File: tomni/cytosmart_data_format/annotations/ellipse/main.py [hunk 2]
```diff
@@ -11,32 +11,89 @@ def __init__(self, radius: Point, center: Point, rotation: float):
...
     @property.getter
     def rotation(self) -> float:
-        """ rotation in degree.
-        """
-        pass
+        return self._rotation
 
     @property.getter
-    def get_circluratiy(self) -> str:
+    def get_circluratiy(self) -> float:
```
[the remark this answers is not in the document]
Comment (schneiderjan) [reply tomni/cytosmart_data_format/annotations/ellipse/main.py:34]: Fixed.

File: tomni/cytosmart_data_format/annotations/ellipse/main.py [hunk 3]
```diff
@@ -11,32 +11,89 @@ def __init__(self, radius: Point, center: Point, rotation: float):
...
+            float: Ellipse's aspect ratio.
+        """
+        if not self._aspect_ratio:
+            self._calculate_aspect_ratio()
+
+        return self._aspect_ratio
+
+    def _calculate_circularity(self) -> None:
+        if not self._perimeter:
+            self._calculate_perimeter()
```
[the remark this answers is not in the document]
Comment (schneiderjan) [reply tomni/cytosmart_data_format/annotations/ellipse/main.py:83]: > 

This not recalculate if value is present.

[the remark this answers is not in the document]
Comment (wagenrace) [reply tomni/cytosmart_data_format/annotations/ellipse/main.py:83]: Yes it does

File: tomni/cytosmart_data_format/annotations/ellipse/main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wagenrace) [reply tomni/cytosmart_data_format/annotations/ellipse/main.py:34]: I think you forgot to push

File: integration_cdf.py [hunk 4]
```diff
@@ -1,3 +1,15 @@
-from .tomni.cytosmart_data_format import CytoSmartDataFormat
+#%%
+from tomni.cytosmart_data_format.annotations import Ellipse, Point
 
-dummy = CytoSmartDataFormat([{}])
+# Ellipse annotation
+elli = Ellipse(
+    radius=Point(1, 3),
```
Comment (wagenrace) [inline integration_cdf.py:6]: This line shows perfectly why the radius of the type point is weird.

File: tomni/cytosmart_data_format/annotations/ellipse/main.py [hunk 5]
```diff
@@ -1,42 +1,133 @@
...
 
 from ..annotation import Annotation
 from ..point import Point
 
 
 class Ellipse(Annotation):
-    def __init__(self, radius: Point, center: Point, rotation: float):
+    def __init__(
+        self,
+        radius: Point,
```
Comment (wagenrace) [inline tomni/cytosmart_data_format/annotations/ellipse/main.py:13]: Replace with 2 parameters; radius x and radius y
Or with radius object

File: tomni/cytosmart_data_format/annotations/ellipse/main.py [hunk 6]
```diff
@@ -1,42 +1,133 @@
...
+        center: Point,
+        rotation: float,
+        id: str,
+        label: str,
+        children: List[Annotation],
+        parents: List[Annotation],
+    ):
```
Comment (wagenrace) [inline tomni/cytosmart_data_format/annotations/ellipse/main.py:20]: add docstring

Comment (wagenrace) [reply tomni/cytosmart_data_format/annotations/ellipse/main.py:20]: Mainly make sure the units are known. Is rotation is fraction, radii, or degrees

File: tomni/cytosmart_data_format/annotations/ellipse/main.py [hunk 7]
```diff
@@ -1,42 +1,133 @@
...
+            self._calculate_aspect_ratio()
+
+        return self._aspect_ratio
+
+    def to_dict(self) -> dict:
+
+        dict_ellipse = {
+            "type": "ellipse",
+            "center": asdict(self.center),
+            "radius": asdict(self.radius),
```
Comment (wagenrace) [inline tomni/cytosmart_data_format/annotations/ellipse/main.py:104]: You are changing the Standard format of the JSON!!!!
Please don't unless you discussed it

File: tomni/cytosmart_data_format/annotations/ellipse/main.py (hunk 5 above)
Comment (schneiderjan) [reply tomni/cytosmart_data_format/annotations/ellipse/main.py:13]: Tom, bram and I were discussing this. Atm we only have `circles` that are ellipses but for future we should also consider how we use elllipses, so let's talk about how we ellipses.

Comment (wagenrace) [reply tomni/cytosmart_data_format/annotations/ellipse/main.py:13]: Let discuss it Wednesday

**Review by wagenrace — APPROVED** [review]

**Pull request merged**